internal/gitinterface/log_test.go

```diff
@@ -102,4 +98,182 @@ func TestGetCommitsBetweenRange(t *testing.T) {
 		})
 		assert.Equal(t, expectedCommits, commits)
 	})
+
+	t.Run("Get commits from invalid range", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, plumbing.ZeroHash, plumbing.ZeroHash)
+		assert.ErrorIs(t, err, plumbing.ErrObjectNotFound)
+		assert.Nil(t, commits)
+	})
+	t.Run("Get commits from non-existent commit", func(t *testing.T) {
+		nonExistentHash := EmptyBlob()
+		commits, err := GetCommitsBetweenRange(repo, nonExistentHash, plumbing.ZeroHash)
+		assert.Nil(t, err)
+		assert.Equal(t, commits, []*object.Commit{})
+	})
+}
+
+func TestGetCommitsBetweenRangeForMergeCommits(t *testing.T) {
+	// Creating a tree with merge commits
+	commitIDs := make([]plumbing.Hash, 0, 6)
+	repo, err := git.Init(memory.NewStorage(), memfs.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	refName := plumbing.ReferenceName("refs/heads/main")
+
+	if err := repo.Storer.SetReference(plumbing.NewHashReference(refName, plumbing.ZeroHash)); err != nil {
+		t.Fatal(err)
+	}
+
+	ref, err := repo.Reference(refName, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	emptyBlobHash, err := WriteBlob(repo, []byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	treeHashes, err := createTreeHashes(repo, emptyBlobHash, 6)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// creating the first commit
+
+	commit := CreateCommitObject(testGitConfig, treeHashes[0], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", 1), testClock)
+
+	applyCommitAndRefreshRef(t, repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating two children from the first commit
+
+	childrenCommitIds, children := createChildrenCommits(t, repo, ref, treeHashes, 2)
+
+	commitIDs = append(commitIDs, childrenCommitIds...)
+
+	// creating a merge commit from the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[4], children, fmt.Sprintf("Test commit %v", 5), testClock)
+
+	applyCommitAndRefreshRef(t, repo, commit, ref)
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating children for each of the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[3], []plumbing.Hash{children[0]}, fmt.Sprintf("Test commit %v", 4), testClock)
```

Comment:
Could you indicate in a comment what each index in `children` maps to in the diagram? It's hard to keep track atm.

Reply:
I did this, and just to clear things up, I made a test for every single commit, so people can understand what connects to what, and how it works.